File: app/src/main/java/dev/expositopablo/tonedeath/main/MainActivity.java
package dev.expositopablo.tonedeath.main;

import android.app.ActivityOptions;
import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.CardView;
import android.text.SpannableString;
import android.text.style.StyleSpan;
import android.view.View;
import android.widget.TextView;

import com.crashlytics.android.Crashlytics;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import dagger.android.AndroidInjection;
import dev.expositopablo.tonedeath.BuildConfig;
import dev.expositopablo.tonedeath.R;
import dev.expositopablo.tonedeath.data.db.DataManager;
import dev.expositopablo.tonedeath.learning.InitFinalListActivity;
import dev.expositopablo.tonedeath.practice.PracticeActivity;
import io.fabric.sdk.android.Fabric;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Inject
    protected DataManager dataManager;

    @BindView(R.id.textView_main_score)
    protected TextView score;
    @BindView(R.id.button_main_learning)
    protected CardView learningButton;
    @BindView(R.id.button_main_practice)
    protected CardView practiceButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        if (!BuildConfig.DEBUG)
            Fabric.with(this, new Crashlytics());

        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        learningButton.setOnClickListener(this);
        practiceButton.setOnClickListener(this);

        Integer rowCount = dataManager.getRowCount().getValue();
        if (rowCount == null || rowCount == 0){
            dataManager.initDB();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        displayScore();
    }

    private void displayScore() {
        int scoreNumber = dataManager.getScore();
        String displayScoreText = getString(R.string.display_score);
        SpannableString span = new SpannableString(displayScoreText + " " + scoreNumber);
        span.setSpan(new StyleSpan(Typeface.BOLD), 0, displayScoreText.length(), 0);
        score.setText(span);
    }

    @Override
    public void onClick(View view) {
        Bundle bundleTransition = ActivityOptions.makeSceneTransitionAnimation(this).toBundle();
        switch (view.getId()) {
            case R.id.button_main_learning:
                startActivity(new Intent(this, InitFinalListActivity.class), bundleTransition);
                break;
            case R.id.button_main_practice:
                startActivity(new Intent(this, PracticeActivity.class), bundleTransition);
                break;
            default: //Nothing
        }
    }
}

File: app/src/main/java/dev/expositopablo/tonedeath/views/main/MainActivity.kt
package dev.expositopablo.tonedeath.views.main

import android.app.ActivityOptions
import android.content.Context
import android.content.Intent
import android.graphics.Typeface
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.text.SpannableString
import android.text.style.StyleSpan
import android.view.View
import androidx.core.view.isVisible
import androidx.lifecycle.Observer
import dev.expositopablo.tonedeath.R
import dev.expositopablo.tonedeath.data.commons.State
import dev.expositopablo.tonedeath.views.BaseActivity
import dev.expositopablo.tonedeath.views.practice.PracticeActivity
import org.koin.android.viewmodel.ext.android.viewModel
import org.koin.core.context.loadKoinModules
import org.koin.core.context.unloadKoinModules
import kotlinx.android.synthetic.main.activity_main.button_main_learning as learningButton
import kotlinx.android.synthetic.main.activity_main.button_main_practice as practiceButton
import kotlinx.android.synthetic.main.activity_main.textView_main_score as score

fun Context.MainActivityIntent() = Intent(this, MainActivity::class.java)

class MainActivity : BaseActivity() {

    private val viewModel by viewModel<MainViewModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        loadKoinModules(mainModule)
        learningButton.setOnClickListener {

        }
        practiceButton.setOnClickListener {
            val bundleTransition = ActivityOptions.makeSceneTransitionAnimation(this).toBundle()
            startActivity(Intent(this, PracticeActivity::class.java), bundleTransition)
        }
        viewModel.state.observe(this, Observer {
            handleState(it)
        })
    }

    override fun onStop() {
        unloadKoinModules(mainModule)
        super.onStop()
    }

    private fun handleState(state: State) {
        when (state) {
            is State.ScoreSucess -> {
                displayScore(state.score)
            }
        }
    }


    private fun displayScore(scoreNumber: Int = 0) {
        score.isVisible = true
        val displayScoreText = getString(R.string.display_score)
        val span = SpannableString("$displayScoreText $scoreNumber")
        span.setSpan(StyleSpan(Typeface.BOLD), 0, displayScoreText.length, 0)
        score.text = span
    }

}
